oak/server/rust/oak_loader/src/main.rs

```diff
@@ -63,9 +68,45 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     };
 
     // Spawns a new thread corresponding to an initial Wasm Oak node.
-    configure_and_run(app_config).map_err(|error| format!("Runtime error: {:?}", error))?;
+    configure_and_run(app_config)
+        .map_err(|error| format!("Runtime error: {:?}", error))
+        .expect("Error when starting the Runtime");
+
     // Park current thread.
     park();
 
     Ok(())
 }
+
+fn start_metrics() {
+    let opt = Opt::from_args();
+    let port = opt.metrics_port.unwrap_or(3030);
+
+    let mut tokio_runtime = tokio::runtime::Runtime::new().expect("Couldn't create Tokio runtime");
+    let result = tokio_runtime.block_on(metrics::serve_metrics(port));
+
+    info!("Exiting metrics server node thread {:?}", result);
+}
+
+fn main() {
+    env_logger::init();
+    let mut handles = vec![];
+
+    // start the runtime in a new thread
+    handles.push(spawn(move || {
+        if let Err(e) = start_runtime() {
+            error!("Error in runtime: {}", e);
+        }
+    }));
+
+    // start metrics server in a new thread
+    handles.push(spawn(move || {
+        start_metrics();
+    }));
+
+    for handle in handles {
```

Comment:
There are only two handles, should they just be two distinct variables for clarity?

Reply:
I prefer this one.